### PR TITLE
[Shader Graph] Fix Various Bugs 

### PR DESCRIPTION
**Pull request by alindmanUnity**
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?

- `Enable GPU Instancing` no longer produces errors when enabled on a Shader Graph material.
FB Case 1184870 [[private link]](https://fogbugz.unity3d.com/f/cases/1184870/) [[public link]](https://issuetracker.unity3d.com/issues/universalrp-shader-compilation-error-when-using-gpu-instancing)
- Multiple tangent transform nodes on a graph no longer produces errors.
FB case 1185752 [[private link]](https://fogbugz.unity3d.com/f/cases/1185752/) [[public link]](https://issuetracker.unity3d.com/issues/shadergraph-fails-to-compile-with-redefinition-of-transposetangent-when-multiple-tangent-transform-nodes-are-plugged-in)
- The Main Preview for Sprite Unlit Sprite Lit master nodes now display the correct color.
FB case 1184656 [[private link]](https://fogbugz.unity3d.com/f/cases/1184656/) [[public link]](https://issuetracker.unity3d.com/issues/shadergraph-preview-for-lit-and-unlit-master-node-wrong-color-when-color-is-set-directly-on-master-node)
- Shader Graph shaders in `Always Included Shaders` no longer crash builds.
FB Case 1191757 [[private link]](https://fogbugz.unity3d.com/f/cases/1191757/) [[public link]](https://issuetracker.unity3d.com/issues/lwrp-build-crashes-when-built-with-shadergraph-file-added-to-always-include-shaders-list)
- The `Transform` node now correctly transforms Absolute World to Object. 
No case filed -- original PR: #4742 

---
### Testing status

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [x] Built a player
- [x] C# and shader warnings (supress shader cache to see them)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/sg%252Fbug-bash-oct-19

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.universal/Editor/ShaderGraph/SubShaders/UniversalPBRSubShader.cs [hunk 1]
```diff
@@ -71,12 +71,10 @@ class UniversalPBRSubShader : IPBRSubShader
             includes = new List<string>()
             {
                 "Packages/com.unity.render-pipelines.core/ShaderLibrary/Color.hlsl",
-                "Packages/com.unity.render-pipelines.core/ShaderLibrary/UnityInstancing.hlsl",
                 "Packages/com.unity.render-pipelines.universal/ShaderLibrary/Core.hlsl",
                 "Packages/com.unity.render-pipelines.universal/ShaderLibrary/Lighting.hlsl",
                 "Packages/com.unity.render-pipelines.universal/ShaderLibrary/Shadows.hlsl",
                 "Packages/com.unity.render-pipelines.universal/ShaderLibrary/ShaderGraphFunctions.hlsl",
-                "Packages/com.unity.render-pipelines.universal/ShaderLibrary/Input.hlsl",
```
Comment (alindmanUnity) [inline com.unity.render-pipelines.universal/Editor/ShaderGraph/SubShaders/UniversalPBRSubShader.cs]: deleted includes here were the cause of GPU instancing errors. they were being included in multiple places down the chain and throwing redefinition errors.

**Comment by zadi0648** [comment]
Where is the mentioned project for https://fogbugz.unity3d.com/f/cases/1191757/ ? I see a link to a .7z project got removed in the description

**Comment by zadi0648** [comment]
1184870 is good ✔️

File: com.unity.shadergraph/CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,11 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Fixed a shader compatibility issue in the SRP Batcher when you use a hybrid instancing custom variable.
 - Fixed an issue where Unity would crash when you imported a Shader Graph Asset with invalid formatting.
 - Fixed an issue with the animated preview when there is no Camera with animated Materials in the Editor.
+- `Enable GPU Instancing` no longer produces errors when enabled on a Shader Graph material.
+- Multiple tangent transform nodes on a graph no longer produces errors.
+- The Main Preview for Sprite Unlit Sprite Lit master nodes now display the correct color.
+- Shader Graph shaders in `Always Included Shaders` no longer crash builds.
```
Comment (phi-lira) [inline com.unity.shadergraph/CHANGELOG.md]: Consider addding the public case link to changelogs. This helps users.

**Comment by LandonTownsendUnity** [comment]
Confirmed 1184656 as fixed

**Comment by LandonTownsendUnity** [comment]
Confirmed the last bugfix (port from PR #4742)

**Comment by LandonTownsendUnity** [comment]
1185752 appears to be fully fixed now.

**Comment by ROBYER1** [comment]
Using Commit f50395ad68e0ec11630bfecafa6e7bdd96ab93fc  from 31 October, the Unity Sample scene using 1 shader graph won't build for VR (Android/Oculus Quest/Go) due to various stereoTargetEyeIndex errors. See Screenshot attached.

Would it be simple to suggest a CI test that tests just building a sample scene with a shader graph in to test anything being merged?

Reported at Case 1195207
stereoTargetEyeIndex Error

**Comment by ROBYER1** [comment]
Things are getting worse, Commit a66889edff19bf58cbba0c02332d92b36cb352aa 
has a warped right eye, UI Worldspace canvas and shader graph does not render correctly.

Tons of OpenGL errors too see the log..

Reported at Case 1195463
everythingisnotinfe

**Comment by LandonTownsendUnity** [comment]
How is the backport of this PR going?

**Comment by soraryu** [comment]
@alindmanUnity I spent the last two hours bisecting why 7.1.5 breaks ShaderGraph on Quest again (even without GPU Instancing on). 

The changes in this PR are the breaking point:

7406d6d898199984b579f5e75eaf31c0b27666ee works (8 Oct 2019)
58911ca88cdff5f5841f3ae25afe8e540bf5c39a works (9 Oct 2019)
c7baf6ee2b83985c77fd2ac2e14ecb252f7c1535 works (11 Oct 2019)
9d6736a40d1f373c26d64834761b8fcdd0ed8299 works (25 Oct 2019)
df943eafe65c0f10dfb5e9c661547b3bb336211c works (29 Oct 2019)
**e962c7ee04f14728c99ef0cf8742a3fb7e816007 breaks (29 Oct 2019)**
4a03f0249c412baa8cf446fdd03d3b33062c815d does not work (4 Nov 2019)
60109fed3d2109829f314c327dd418025bb7f941 does not work (15 Nov 2019)
54221e0fde249f375d64165477abdf237faee480 does not work (31 Oct 2019)

**Pull request merged**